src/lib/Hydra/Controller/API.pm

```diff
@@ -87,7 +87,8 @@ sub jobsetToHash {
         checkinterval => $jobset->checkinterval,
         triggertime => $jobset->triggertime,
         fetcherrormsg => $jobset->fetcherrormsg,
-        errortime => $jobset->errortime
+        errortime => $jobset->errortime,
+        successful => $jobset->errormsg eq "" ? JSON::true : JSON::false
```

Comment:
Does this query always fetch the entire error message every time? Did it before?

Does `errormsg` mean `stderr` output?

If so, the presence  of stderr output doesn't mean it failed which makes "successful" a misnomer.

Reply:
Actually the semantic of `successful` is not correct: when a fetch error occurs, we don't want to have a `successful` attribute set to `true`.

So, we propose `has_errormsg` attribute instead.